contracts/contracts/ocr2/SequencerUptimeFeed/library.cairo

```diff
@@ -0,0 +1,250 @@
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin, SignatureBuiltin
+from starkware.starknet.common.syscalls import (
+    get_tx_info,
+    get_block_timestamp,
+    get_caller_address,
+    get_block_number,
+)
+from starkware.cairo.common.math import assert_not_zero, assert_le
+from starkware.cairo.common.math_cmp import is_le
+from starkware.cairo.common.bool import TRUE, FALSE
+
+from utils import assert_boolean
+from ocr2.interfaces.IAggregator import Round, AnswerUpdated, NewRound
+from SimpleReadAccessController.library import simple_read_access_controller
+from ownable import Ownable_only_owner
+
+@event
+func RoundUpdated(status : felt, transmission_timestamp : felt):
+end
+
+@event
+func UpdateIgnored(
+    latest_status : felt, latest_timestamp : felt, incoming_status : felt, incoming_timestamp : felt
+):
+end
+
+@event
+func L1SenderTransferred(prev : felt, cur : felt):
+end
+
+@storage_var
+func s_l1_sender() -> (address : felt):
+end
+
+@storage_var
+func s_l2_cross_domain_messenger() -> (address : felt):
+end
+
+@storage_var
+func s_rounds(id : felt, field : felt) -> (res : felt):
+end
+
+@storage_var
+func s_latest_round_id() -> (res : felt):
+end
```

Comment:
I think `library` should always namespace storage variables (`SequencerUptimeFeed_latest_round_id`) etc. to avoid name collisions if it's imported elsewhere. I saw this pattern from OpenZeppelin's `library.cairo` implementations

Reply:
I'm not sure that this is necessary. I think if child class has intersecting variable names then probably smth off with architecture or those variables named improperly incorrectly reflecting their meaning. If this is for the case where we make functions from library reusable not in class that inherit from it then such functions shall be placed in `utils.cairo`

Reply:
Namespace examples: https://github.com/OpenZeppelin/cairo-contracts/blob/main/src/openzeppelin/access/accesscontrol/library.cairo

@archseer please make a decision here. We should be consistent across our codebase.

@taco-paco these `s_` prefixes are c/p from our Solidity implementation, right?

Reply:
Saw this somewhere in other cairo contracts. This just lets know that this is storage_var from name + solves issues with redefinition like
`let (latest_round_id) = s_latest_round_id.read()`

Reply:
I think we should follow the OpenZeppelin guidelines: https://docs.openzeppelin.com/contracts-cairo/0.3.1/extensibility#libraries

> Must prefix storage functions with the name of the namespace to prevent clashing with other libraries (e.g. ERC20balances)

I agree that properly named variables are a good practice, but without namespacing subtle bugs can be introduced since it won't actually error, it'll just behave incorrectly.

Reply:
Discussed on the meeting at 18.8. and agreed?